autonomy/analyse/dialogues.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Analyse dialogue definitions."""
+
+
+import importlib
+import importlib.machinery
+import importlib.util
+import types
+from importlib.machinery import ModuleSpec
+from pathlib import Path
+from typing import Any, Dict, List, cast
+
+import yaml
+
+from autonomy.analyse.constants import CLASS_NAME, DIALOGUES, DIALOGUES_FILE, MODELS
+
+
+DIALOGUES_LOWER = DIALOGUES.lower()
+
+
+def load_dialogues_module_from_skill_path(skill_path: Path) -> types.ModuleType:
+    """Load `dialogues.py` module for the given skill."""
+
+    loader = importlib.machinery.SourceFileLoader(
+        DIALOGUES, str(skill_path / DIALOGUES_FILE)
+    )
+    spec = cast(ModuleSpec, importlib.util.spec_from_loader(DIALOGUES, loader))
+    module = importlib.util.module_from_spec(spec)
+    loader.exec_module(module)
+
+    return module
+
+
+def validate_and_get_dialogues(configuration: Dict[str, Any]) -> Dict[str, str]:
+    """Returns dialogue names to class name mappings"""
+
+    mapping = {}
+    for name, info in cast(Dict[str, Dict[str, str]], configuration[MODELS]).items():
+        name_ends_with_dialogue = name.endswith(DIALOGUES_LOWER)
+
+        class_name = info[CLASS_NAME]
+        class_ends_with_dialogue = class_name.lower().endswith(DIALOGUES_LOWER)
+
+        if not class_ends_with_dialogue and not name_ends_with_dialogue:
+            continue
+
+        if name_ends_with_dialogue and not class_ends_with_dialogue:
+            raise ValueError(
+                f"Class name for a dialogue should end with `Dialogues`; dialogue={name}; class={class_name}"
+            )
+
+        if class_ends_with_dialogue and not name_ends_with_dialogue:
+            raise ValueError(
+                f"Dialogue name should end with `dialogues`; dialogue={name}; class={class_name}"
+            )
```

Comment:
Is using a lower case string to perform checks enough or should we make them case sensitive?

Reply:
Should be fine, we have linters for checking the naming conventions.